### PR TITLE
AvatarPopover: Use a FileIcon for filechooser previews

### PR DESCRIPTION
**Pull request by danrabbit**
Supersedes #104 

Use a FileIcon so we get free HiDPI support. Also fixes previews for files with spaces in the name.

### CODE REVIEW

**Comment by torikulhabib** [comment]
Screen record from 2019-04-08 18 26 17

**Comment by torikulhabib** [comment]
if not delete this.
file_dialog.set_preview_widget_active (true);	
preview is working.

File: src/Widgets/Popovers/AvatarPopover.vala [hunk 1]
```diff
@@ -78,20 +80,11 @@ namespace SwitchboardPlugUserAccounts.Widgets {
...
-                        Gdk.Pixbuf pixbuf = new Gdk.Pixbuf.from_file_at_scale (uri.substring (7), 150, 150, true);
-                        preview_area.set_from_pixbuf (pixbuf);
-                        preview_area.show ();
-                        file_dialog.set_preview_widget_active (true);
-                    } catch (Error e) {
-                        preview_area.hide ();
-                        file_dialog.set_preview_widget_active (false);
-                    }
+                if (uri != null && uri.has_prefix ("file://")) {
+                    preview_area.set_from_gicon_async.begin (new FileIcon (file_dialog.get_file ()), 256);
```
Comment (jeremypw) [inline src/Widgets/Popovers/AvatarPopover.vala]: Unhandled error may be thrown.  Should hide preview area if image not successfully loaded (or file is not an image file) in order to maintain current behaviour.

**Comment by jeremypw** [comment]
@danrabbit  Are you in a position to progress this?  You can handle the potential error thrown by AsyncImage.set_from_gicon_async () like this:
```
                if (uri != null && uri.has_prefix ("file://")) {
                    preview_area.set_from_gicon_async.begin (
                        new FileIcon (file_dialog.get_file ()),
                        256,
                        null,
                        (obj, res) => {
                            try {
                                preview_area.set_from_gicon_async.end (res);
                                preview_area.show ();
                            } catch (Error e) {
                                preview_area.hide ();
                            }
                        });
                } else {
                    preview_area.hide ();
                }
```

Although I am not sure whether the error could ever be thrown in practice.

**Comment by danrabbit** [comment]
@jeremypw sorry I missed this! Updated

**Review by jeremypw — APPROVED** [review]

**Pull request merged**